packages/repository/src/repositories/repository.ts

```diff
@@ -250,16 +251,18 @@ export class CrudRepositoryImpl<T extends Entity, ID>
     return this.toModels(this.connector.find(this.model, filter, options));
   }
 
-  findById(id: ID, options?: Options): Promise<T> {
+  async findById(id: ID, filter?: Filter, options?: Options): Promise<T> {
```

Comment:
What filter would `findById` method possibly take? I think the interface should be updated to remove this method instead of adding it to the implementation. 

If it is supposed to be here then can you add the `@param` annotation to the interface since it is missing.

Reply:
I am also curious about the filter for `findById`.

Reply:
If you have a `Customer` model that has many `Order` instances, then you can fetch a customer with all their orders.

```
GET /api/customers/123?filter[include]=orders
```

AFAIK, juggler supports only a subset of Filter properties. For example, `fields` and `include` are most likely supported, but `where` is not.

On the second thought, using the `Filter` type was a bad idea, we should use a different type that describes only the properties applicable to `findById`. Let's leave this change out of this initial pull request. I opened a new issue to keep track of it, see https://github.com/strongloop/loopback-next/issues/1721

Reply:
Never mind. The `filter` argument of `findById` has been added as part of the initial repository implementation in e92cc244a.

Removing `filter` argument from `findById` signature is a change not related to handling "not found" situation.

I think this particular source code line is highlighted by git because previously, the default repository implementation (that nobody uses) did not have `filter` argument and as I was fixing `findById` signatures, I made the signature consistent with the rest.

I'll add a minimal `@param` annotation to the interface then.